### PR TITLE
MES-8718 | Analytics for comms selection

### PR DESCRIPTION
**Pull request by gittins7**
## Description
GA for comms selection
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/pages/communication/communication.actions.ts [hunk 1]
```diff
@@ -17,3 +17,15 @@ export const CommunicationValidationError = createAction(
...
+  '[WaitingRoomToCarPage] Booking email selected',
+);
+
+export const NewEmailSelected = createAction(
+  '[WaitingRoomToCarPage] New email selected',
+);
+
+export const PostalSelected = createAction(
+  '[WaitingRoomToCarPage] Post selected',
+);
```
Comment (matthew2564) [inline src/app/pages/communication/communication.actions.ts]: Please change prefixes to `[CommunicationPage]`

File: src/app/pages/communication/communication.analytics.effects.ts [hunk 2]
```diff
@@ -162,6 +167,42 @@ export class CommunicationAnalyticsEffects {
     }),
   ));
 
+  bookingEmailSelected$ = createEffect(() => this.actions$.pipe(
+    ofType(BookingEmailSelected),
+    switchMap(() => {
+      this.analytics.logEvent(
+        AnalyticsEventCategories.WAITING_ROOM_TO_CAR,
```
Comment (matthew2564) [inline src/app/pages/communication/communication.analytics.effects.ts]: Do these not need to be the comms page?

File: src/app/pages/communication/components/new-email/new-email.ts [hunk 3]
```diff
@@ -10,6 +13,11 @@ import { UntypedFormGroup, UntypedFormControl, Validators } from '@angular/forms
 })
 export class NewEmailComponent implements OnChanges {
 
+  constructor(
+    private store$: Store<StoreModel>,
+  ) {
+  }
+
```
Comment (matthew2564) [inline src/app/pages/communication/components/new-email/new-email.ts]: We should not do this here, we should dispatch an action on the emission of `newEmailRadioSelect` in the parent i.e. the comms page

File: src/app/pages/communication/components/postal-address/postal-address.ts [hunk 4]
```diff
@@ -11,6 +14,11 @@ import { UntypedFormGroup } from '@angular/forms';
 })
 export class PostalAddressComponent {
 
+  constructor(
+    private store$: Store<StoreModel>,
+  ) {
+  }
+
```
Comment (matthew2564) [inline src/app/pages/communication/components/postal-address/postal-address.ts]: Same as above

File: src/app/pages/communication/components/provided-email/provided-email.ts [hunk 5]
```diff
@@ -10,6 +13,11 @@ import { UntypedFormControl, UntypedFormGroup, Validators } from '@angular/forms
 })
 export class ProvidedEmailComponent {
 
+  constructor(
+    private store$: Store<StoreModel>,
+  ) {
+  }
+
```
Comment (matthew2564) [inline src/app/pages/communication/components/provided-email/provided-email.ts]: Same as above

**Review by matthew2564 — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]

**Pull request merged**